src/Magnum/ImGuiIntegration/Context.hpp

```diff
@@ -202,6 +202,63 @@ template<class TextInputEvent> bool Context::handleTextInputEvent(TextInputEvent
     return false;
 }
 
+/* GLFW doesn't have all cursor types, so employing a dirty SFINAE trick that
+   falls back to Arrow if given enum value is not available */
+namespace Implementation {
+
+#ifndef DOXYGEN_GENERATING_OUTPUT
+#define MAGNUM_IMGUIINTEGRATION_OPTIONAL_CURSOR(cursor)                     \
+    template<class U> struct Optional##cursor##Cursor {                     \
+        template<class T> constexpr static typename T::Cursor get(T*, decltype(T::Cursor::cursor)* = nullptr) { return T::Cursor::cursor; } \
+        constexpr static typename U::Cursor get(...) { return U::Cursor::Arrow; } \
+        public:                                                             \
+            constexpr static typename U::Cursor Cursor = get(static_cast<U*>(nullptr)); \
+    };
+MAGNUM_IMGUIINTEGRATION_OPTIONAL_CURSOR(ResizeAll)
+MAGNUM_IMGUIINTEGRATION_OPTIONAL_CURSOR(ResizeNESW)
+MAGNUM_IMGUIINTEGRATION_OPTIONAL_CURSOR(ResizeNWSE)
+#undef MAGNUM_IMGUIINTEGRATION_OPTIONAL_CURSOR
+#endif
+
+}
+
+template<class Application> void Context::updateApplicationCursor(Application& application) {
+    /* Ensure we use the context we're linked to */
+    ImGui::SetCurrentContext(_context);
+
+    switch(ImGui::GetMouseCursor()) {
+        case ImGuiMouseCursor_TextInput:
+            application.setCursor(Application::Cursor::TextInput);
```

Comment:
*Application::Cursor::Text

Or maybe you prefer TextInput as name and this is a way to telling me? 👀

Reply:
Argh, yes, when seeing that naming in ImGui it immediately looked better than CSS's `text` and SDL/GLFW's `IBeam`. But of course I forgot to mention that when doing the final review on your PR :see_no_evil: